train_runner.py

```diff
@@ -7,71 +7,115 @@
+                pass
+            else:
+                print("%s = %r" % (attr, getattr(self, attr)))
+
+params = Params()
+params.pprint()
+
+TEXT, word_embeddings, train_iter, test_iter = load_dataset(data_source=params.DATASET_NAME, fix_length=params.MAX_LEN,
+                                                            device=params.device, batch_size_train=params.TRAIN_BATCH_SIZE,
+                                                            batch_size_test=params.TEST_BATCH_SIZE)
+
+print('Train dataset len: {} Test dataset len: {}'.format(len(train_iter.dataset), len(test_iter.dataset)))
 
 # Clear CUDA memory if needed
 # TODO: local use
 # torch.cuda.empty_cache()
 
 ### Init models ###
-device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
-print('Device is {}'.format(device))
 
-vocab_size = len(tokenizer.vocab)
-rm = ResourcesManager(vocab_size, MODELS_PATH, "basic", device, load_path=MODELS_PATH)
+vocab_size = len(TEXT.vocab)
+rm = ResourcesManager(vocab_size, params.MODELS_PATH, "basic_exp_{}".format(params.EXP_NUM),
+                      params, load_path=params.MODELS_PATH)
 models = rm.get_models()
 model_enc = models["enc"]
 model_dec = models["dec"]
 model_cls = models["cls"]
 
+# TODO: if using this option set H_DIM = 300
+# model_enc = load_pretrained_embedding_to_encoder(model_enc, word_embeddings)
+
 ### Init losses ###
 cls_criteria = nn.CrossEntropyLoss()
 seq2seq_criteria = LabelSmoothing(size=vocab_size, padding_idx=0)
 ent_criteria = EntropyLoss()
 
 ### Init optimizers ###
-cls_opt = NoamOpt(H_DIM, 2, OPT_WARMUP_FACTOR,
-            torch.optim.Adam(model_cls.parameters(), lr=LR, betas=(0.9, 0.98), eps=1e-9))
-enc_opt = get_std_opt(model_enc)
-dec_opt = get_std_opt(model_dec)
+cls_opt = get_std_opt(model_cls, params)
+enc_opt = get_std_opt(model_enc, params)
+dec_opt = get_std_opt(model_dec, params)
 
-model_enc.train()
-model_dec.train()
-model_cls.train()
+early_stop = EarlyStopping(params.PATIENCE)
+for epoch in range(Params.N_EPOCHS):
+  rm.on_epoch_end(epoch)
 
-for epoch in range(N_EPOCHS):
-  run_epoch(epoch, rm, train_dataloader, model_enc, enc_opt, model_dec, dec_opt,
+  run_epoch(epoch, rm, train_iter, model_enc, enc_opt, model_dec, dec_opt,
               model_cls, cls_opt, cls_criteria, seq2seq_criteria,
-              ent_criteria, device, trans_steps=TRANS_STEPS, cls_steps=CLS_STEPS,
-              rec_lambda=REC_LAMBDA, print_interval=PRINT_INTERVAL, verbose=VERBOSE)
+              ent_criteria, params.device, params=params)
+
+  rm.on_epoch_end(epoch)
+  test_acc = evaluate(epoch, test_iter, model_enc, model_dec,
+             model_cls, cls_criteria, seq2seq_criteria,
+             ent_criteria, params.device)
+
+  test_random_samples(test_iter, TEXT, model_enc, model_dec, model_cls, params.device,
+                      decode_func=greedy_decode_sent, num_samples=2, transfer_style=True)
+
+  # TODO - Roy - currently not in use, what metric to follow ?
```

Comment:
I think we should use accuracy

Reply:
we have two measures of acc - reconstruction and classifications, with one ? an aggregation of the two ?